### PR TITLE
Allow disabling datewidget

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description

The `<DateWidget>` wasn't able to be disabled. With this PR, the widget can now be disabled by setting `ui:disabled` in the `uiSchema`.

Context: https://dsva.slack.com/archives/CBU0KDSB1/p1608134523456600

## Testing done

Unit testing & browser testing

## Screenshots

The `<select>`s are actually disabled even though they don't look like it. This may be an opportunity to improve the CSS at some point.

image

## Acceptance criteria
- [ ] `<DateWidget>` can be disabled

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/test/js/widgets/DateWidget.unit.spec.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { expect } from 'chai';
 import SkinDeep from 'skin-deep';
+import { render } from '@testing-library/react';
```
Comment (bkjohnson) [inline src/platform/forms-system/test/js/widgets/DateWidget.unit.spec.jsx:4]: `skin-deep` hasn't been updated in three years, and it has missing documentation. I couldn't find a way to check attributes with it, so I wrote the test using RTL.

**Review by cvalarida — COMMENTED** [review]
This looks good, but I don't think we can ship this without those CSS changes though to make the selects _look_ disabled. This should happen in `formation`, but until then, [`_m-schemaform.scss`](https://github.com/department-of-veterans-affairs/vets-website/tree/master/src/platform/forms/sass/_m-schemaform.scss) is probably the best place..

**Review by ncksllvn — APPROVED** [review]

**Review by stephenBarrs — APPROVED** [review]
Thanks for working on this so quickly.  Looks Great!  CSS updates would be helpful but we could MVP without if necessary.

**Comment by bkjohnson** [comment]
It looks like neither [WDS](https://designsystem.digital.gov/components/form-controls/#dropdown) nor our own design system offers any styling/guidance around what a disabled `<select>` should look like. So, all I did was change the background color:

image

**Comment by stephenBarrs** [comment]
@batemapf I think the background color change is sufficient to communicate `Read Only` for the data widget.

**Review by cvalarida — APPROVED** [review]

**Pull request merged**